parsl/dataflow/job_error_handler.py

```diff
@@ -0,0 +1,49 @@
+from typing import List, Any, Dict
+
+from parsl.dataflow.task_status_poller import ExecutorStatus
+from parsl.executors.base import ParslExecutor
+from parsl.providers.provider_base import JobStatus, JobState
+
+
+class JobErrorHandler(object):
+    def __init__(self):
+        pass
+
+    def run(self, status: List[ExecutorStatus]):
+        for es in status:
+            self._check_irrecoverable_executor(es)
+
+    def _check_irrecoverable_executor(self, es: ExecutorStatus):
+        if not es.executor.error_management_enabled:
+            return
+        custom_handling = es.executor.handle_errors(self, es.status)
+        if not custom_handling:
```

Comment:
maybe there's no need for a `not custom_handling` case here? The StatusHandlingExecutor code has a different default that overrides this default, and (I think?) the NoStatusHandlingExecutor shouldn't hit this code path because error_management_enabled() is false? So "if you say you want error management, you have to handle the errors"?

Reply:
The original idea was that errors can be handled by a common handler with "if 3/3 blocks fail, the executor is bad". That until you realize that if init_blocks=max_blocks=2, you'll never hit 3 failed blocks.  So for executors like that, some special handling is needed. It turns out that all executors that submit "blocks" have init_blocks and max_blocks, so, for the current set of executors what you say is true. But init_blocks and max_blocks is not standardized in any way, so it would be unwise to pretend that all executors inheriting from StatusHandlingExecutors will magically have this working when the error handing in StatusHandlingExecutors works only if hasattr("init_blocks").

Reply:
I think what I'm saying is basically in line with that: don't assume there is any kind of known-to-core-parsl sensible error behaviour and make each executor provide the sensible-for-that-executor behaviour (which it can inherit or delegate if that happens to make sense - but that is up to the executor implementation)

Reply:
Except the initial design, and we can talk separately about the details of how that design was arrived at, called for a known-to-core-parsl error behavior, more in line with the way strategies work. Until it turned out that it's not quite sensible with the current executors. So a compromise solution is there. I don't particularly care, but I also don't see much of a downside in having some not-particularly-not-sensible default behavior and inheritance+delegation, as in this PR.